cmd/libsnap-confine-private/locking.c

```diff
@@ -100,8 +100,13 @@ int sc_lock(const char *scope)
 	}
 	// Construct the name of the lock file.
 	char lock_fname[PATH_MAX] = { 0 };
-	sc_must_snprintf(lock_fname, sizeof lock_fname, "%s/%s.lock",
-			 sc_lock_dir, scope ? : "");
+	if (uid == 0) {
```

Comment:
will root be able to acquire the lock?

Reply:
why not?

Reply:
Because we said we won't allow root user to have a mount namespace that is distinct from that of the current notion of a mount namespace for a given snap. This would be confusing for services that use $SNAP_DATA and don't have $SNAP_USER_DATA.

Reply:
Can you leave a comment in the code then?